prime-angular-frontend/src/app/modules/adjudication/shared/components/enrollee-table/enrollee-table.component.ts

```diff
@@ -118,6 +121,13 @@ export class EnrolleeTableComponent implements OnInit {
     this.hasRenewalDateRange = false;
   }
 
+  public toggleFilterAssigned() {
+    this.hasAssignedToFilter = !this.hasAssignedToFilter;
+    this.dataSource.filter = this.hasAssignedToFilter
+      ? { ...this.form.value, adjudicatorIdir: this.adjudicatorIdir } as string
+      : this.dataSource.filter = { ...this.form.value };
```

Comment:
Feels like could reuse the existing `getFilterPredicate()` to include the filter by `adjudicatorIdir`.  It's cleaner to have all filters in one central place rather than creating individual filtering methods if there are more filters needed in the future.

Reply:
It is using the same getFilterPredicate. This is simply setting what the filter is. Same as we do on form value changes for the filter dates.

Reply:
Sorry for missing the predicate part. I think the toggle of the button and the filter change logic can be separated to be cleaner using a BehaviorSubject<boolean> object you can subscribe changes to :

`public hasAssignedToFilter: BehaviorSubject<boolean>;`

```HTML
  <button mat-icon-button
          #assignFilterButton
          [class]="this.hasAssignedToFilter.getValue() ? 'blueIcon' : ''"
          matTooltip="Filter assigned to me"
          (click)="this.toggleFilterAssigned()">
    <mat-icon>assignment_ind icon</mat-icon>
  </button>
```

```Javascript
 public toggleFilterAssigned() {
    this.hasAssignedToFilter.next(!this.hasAssignedToFilter.value);
  }
```

In initForm you can subscribe to the value change of the `hasAssignedToFilter`
```Javascript
    this.hasAssignedToFilter.subscribe(value => {
      this.dataSource.filter = this.hasAssignedToFilter.getValue()
        ? { ...this.form.value, adjudicatorIdir: this.adjudicatorIdir } as string
        : this.dataSource.filter = { ...this.form.value };
    });


```
Slight change in getFilterPredicate
```Javascript
      if (this.hasAssignedToFilter.getValue()) {
        const searchByIdir = row.adjudicatorIdir === filter.adjudicatorIdir;
        matchFilter.push(searchByIdir);
      }
```